hail/src/main/scala/is/hail/types/physical/PCanonicalArray.scala

```diff
@@ -11,9 +11,12 @@ import is.hail.types.virtual.{TArray, Type}
 import is.hail.utils._
 
 // This is a pointer array, whose byteSize is the size of its pointer
-final case class PCanonicalArray(elementType: PType, required: Boolean = false) extends PArray {
+final case class PCanonicalArray(elementType: PType, required: Boolean = false)
+    extends PArray with PArrayBackedContainer {
```

Comment:
Is there any reason not to make `PArray` extend `PArrayBackedContainer` directly?

Reply:
Let's try it!

Reply:
It works. This is free to be merged.